### DISCUSSION TITLE
LLDP 802.3 BT support

### DISCUSSION THREAD

**Discussion by d1115choi** [lldpd/lldpd#568] [Q&A]
Hello LLDP folks,

I use 1.0.15 lldp stable version. From the source code, I built/run the image on Ubuntu LTS 14.04.
Changenote of 1.0.5 says it supports 802.3 BT mode. 

With the lldpcli, I can generate LLDPDU including "Ieee 802.3 - Power Via MDI" TLV, whose TLV type=127.
This TLV includes TLV header, basic field, DLL classification extention but not type 3 and type 4 extension with the command below:
lldpcli configure dot3 power pd supported enabled powerpairs spare class class-3 type 1 source pse priority low requested 10000 allocated 15000

I am wondering how I can generate LLDPDU including the type 3 and type 4 extention.
Additionally, if LLDPDU does not include the type 3 and type 4 extention" for type 3 or 4, do we violate the standard?

Any help will be highly appreciated.

Regards,
David Choi
![wireshark_capture](https://user-images.githubusercontent.com/18752870/213481740-e9abf29c-e3b3-43d1-bbce-9e7e2cc98a4d.png)

**Comment by Neusch** [comment]
Hello everyone,

I also had the same question that David has. It seems that the newest lldpd version does not support configuring 802.3bt support through lldpcli commands. I checked the manual and do not see bt poe supported listed under the lldpcli configure dot3 power command.

I took a look through the code and never see the set function used for the various 802.3bt parameters such as "lldpctl_k_dot3_power_type_ext." Is there a command that I am missing for enabling 802.3bt dot3 POE-MDI TLV power parameters? Is support for the configuring 802.3bt through lldpcli commands planned in a future version?

I appreciate your time to help answer my question. Thank you!

**Reply by bryceremick** [reply]
@Neusch @d1115choi 

There is a PR that contains this functionality that has not been merged. I know this because I have the same issue.

https://github.com/lldpd/lldpd/pull/449

**Reply by d1115choi** [reply]
Hi Bryceremick,
Thank you so much for your link. I almost give up.
